### PR TITLE
csi: add logrotate file to a csi log dir

### PR DESCRIPTION
**Pull request by parth-gr**
added the csi log to a csi log directory csi_log_${node}

Also updated correct dir path to cllect the ceph daemon logs

closes: https://github.com/red-hat-storage/odf-must-gather/issues/178

### CODE REVIEW

File: collection-scripts/gather_ceph_logs [hunk 1]
```diff
@@ -17,6 +17,17 @@ for ns in $namespaces; do
         oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}"/log "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
     }
 
+    # Collecting csi and ceph daemon logs
+    ceph_daemon_log_collection() {
+        dbglog "collecting csi and ceph daemon logs from node ${node}"
+        # as per the correction we are keeping both /var/lib/rook & /var/log/ceph for ceph pods, as to not break anything
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/log/ceph/ "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
```
Comment (parth-gr) [inline collection-scripts/gather_ceph_logs:24]: once the csi operator changes are done we need to add the query the driver and get the host path here.

**Comment by parth-gr** [comment]
cc @Madhu-1 @travisn

File: collection-scripts/gather_ceph_logs [hunk 2]
```diff
@@ -45,11 +56,12 @@ CMDS
         dbglog "collecting crash, journal and volume logs from node ${node}"
         CRASH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/crash_${node}
         CEPH_DAEMON_LOG_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/ceph_daemon_log_${node}
+        CSI_LOG_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/csi_log_${node}
         JOURNAL_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/journal_${node}
         KERNEL_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/kernel_${node}
         COREDUMP_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/coredump_${node}
         mkdir -p "${CRASH_OUTPUT_DIR}"
-        mkdir -p "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
```
Comment (Madhu-1) [inline collection-scripts/gather_ceph_logs:52]: this should not be removed right?

Comment (parth-gr) [reply collection-scripts/gather_ceph_logs:52]: Correct

File: collection-scripts/gather_ceph_logs (hunk 1 above)
Comment (yati1998) [reply collection-scripts/gather_ceph_logs:24]: ceph daemon logs are collected above right?

Comment (parth-gr) [reply collection-scripts/gather_ceph_logs:24]: The path is incorrect `/var/lib/rook` which should be `/var/log/ceph`

As discussed offline already we will keep both as not break anything

File: collection-scripts/gather_ceph_logs [hunk 3]
```diff
@@ -17,6 +17,17 @@ for ns in $namespaces; do
...
 
+    # Collecting csi and ceph daemon logs
+    csi_log_collection() {
+        dbglog "collecting csi and ceph daemon logs from node ${node}"
+        # as per the correction we are keeping both /var/lib/rook & /var/log/ceph for ceph pods, as to not break anything
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/log/ceph/ "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}".rbd.csi.ceph.com/log "${CSI_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}".cephfs.csi.ceph.com/log "${CSI_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}".nfs.csi.ceph.com/log "${CSI_LOG_OUTPUT_DIR}"
+
```
Comment (black-dragon74) [inline collection-scripts/gather_ceph_logs:28]: nit, remove this extraneous space pls. Rest LGTM.

Comment (parth-gr) [reply collection-scripts/gather_ceph_logs]: Updated

File: collection-scripts/gather_ceph_logs (hunk 1 above)
Comment (black-dragon74) [reply collection-scripts/gather_ceph_logs:24]: This is resolved then, right?

Comment (parth-gr) [reply collection-scripts/gather_ceph_logs:24]: Yes correct, will have a follow up PR once CSI-operator is there, 
Till now it works with Rook operator

**Pull request merged**